src/mountscript.sh

```diff
@@ -8,319 +8,26 @@
 #
-fix_dirty_bytes_config()
+is_valid_fqdn()
 {
-    # Constants for desired settings.
-    local desired_dirty_bytes=$((8 * 1024 * 1024 * 1024))  # 8 GB in bytes
-    local desired_dirty_background_bytes=$((4 * 1024 * 1024 * 1024))  # 4 GB in bytes
-
-    # Get current settings.
-    local current_dirty_bytes=$(cat /proc/sys/vm/dirty_bytes 2>/dev/null)
-    local current_dirty_background_bytes=$(cat /proc/sys/vm/dirty_background_bytes 2>/dev/null)
-
-    # Should not happen but added for robustness.
-    if [ -z "$current_dirty_bytes" -o -z "$current_dirty_background_bytes" ]; then
-        wecho "current_dirty_bytes=$current_dirty_bytes"
-        wecho "current_dirty_background_bytes=$current_dirty_background_bytes"
-        return
-    fi
-
-    # If current dirty bytes are 0, calculate them from the ratio configs.
-    if [ $current_dirty_background_bytes -eq 0 ]; then
-        # Get total memory in bytes.
-        local total_mem_KB=$(grep MemTotal /proc/meminfo | awk '{print $2}')
-        local total_mem_bytes=$((total_mem_KB * 1024))
-
-        local current_dirty_ratio=$(cat /proc/sys/vm/dirty_ratio 2>/dev/null)
-        local current_dirty_background_ratio=$(cat /proc/sys/vm/dirty_background_ratio 2>/dev/null)
-
-        # Should not happen but added for robustness.
-        if [ -z "$current_dirty_ratio" -o \
-             -z "$current_dirty_background_ratio" -o \
-             "$current_dirty_ratio" == "0" -o \
-             "$current_dirty_background_ratio" == "0" ]; then
-            wecho "current_dirty_ratio=$current_dirty_ratio"
-            wecho "current_dirty_background_ratio=$current_dirty_background_ratio"
-            return
-        fi
-
-        current_dirty_bytes=$((total_mem_bytes * current_dirty_ratio / 100))
-        current_dirty_background_bytes=$((total_mem_bytes * current_dirty_background_ratio / 100))
-    fi
-
-    # If current dirty byte settings are higher than desired, set to desired.
-    if [ $desired_dirty_background_bytes -lt $current_dirty_background_bytes ]; then
-        vvecho "Setting /proc/sys/vm/dirty_bytes to $desired_dirty_bytes bytes"
-        echo $desired_dirty_bytes > /proc/sys/vm/dirty_bytes
-
-        vvecho "Setting /proc/sys/vm/dirty_background_bytes to $desired_dirty_background_bytes bytes"
-        echo $desired_dirty_background_bytes > /proc/sys/vm/dirty_background_bytes
-    fi
-}
-
-#
-# Help fix the mount options passed in by user.
-#
-fix_mount_options()
-{
-    matchstr="\<sec\>=([^,]+)"
-    if ! [[ "$MOUNT_OPTIONS" =~ $matchstr ]]; then
-        MOUNT_OPTIONS="$MOUNT_OPTIONS,sec=sys"
+    # If AzureEndpointOverride environment variable is set, use it to verify FQDN
+    if [[ -n "$AzureEndpointOverride" ]]; then
+        ModifiedEndPoint = $(echo $AzureEndpointOverride |  sed 's/\./\\./g')
+        [[ $1 =~ ^([a-z0-9]{3,24}|fs-[a-z0-9]{1,21})(\.z[0-9]+)?(\.privatelink)?\.(file|blob)(\.preprod)?\.core+$ModifiedEndPoint$ ]]
```

Comment:
couple of comments.
Was this used during testing or do we have a real usecase for customers needing AzureEndpointOverride?
If we have missed out something, we can add to the regexp, instead of making it so flexible?

If we decide to use it, pls look at following comments:
1. no spaces around = at L26
2. we use all caps for other env variables, so AZURE_ENDPOINT_OVERRIDE?
3. also, it makes sense to not ask user to start this with a dot, rather add a dot after core and ask user to provide "example.endpoint" instead of ".example.endpoint".